scripts/metrics-agent-setup.ps1

```diff
@@ -0,0 +1,79 @@
+Param(
+    [Parameter(Mandatory=$true)]
+    [string]$MetricsWindowsBinariesURL
+)
+
+$ErrorActionPreference = "Stop"
+
+$utils = (Resolve-Path "$PSScriptRoot\Modules\Utils").Path
+Import-Module $utils
+
+$variables = (Resolve-Path "$PSScriptRoot\variables.ps1").Path
+. $variables
+
+function New-MetricsEnvironment {
+    $service = Get-Service $METRICS_SERVICE_NAME -ErrorAction SilentlyContinue
+    if($service) {
+        Stop-Service -Force -Name $METRICS_SERVICE_NAME
+        & sc.exe delete $METRICS_SERVICE_NAME
+        if($LASTEXITCODE) {
+            Throw "Failed to delete exiting $METRICS_SERVICE_NAME service"
+        }
+        Write-Output "Deleted existing $METRICS_SERVICE_NAME service"
+    }
+    New-Directory -RemoveExisting $METRICS_DIR
+    New-Directory $METRICS_BIN_DIR
+    New-Directory $METRICS_LOG_DIR
+    New-Directory $METRICS_SERVICE_DIR
+    New-Directory $METRICS_CONFIG_DIR
+}
+
+function Install-MetricsFiles {
+    $filesPath = Join-Path $env:TEMP "metrics-files.zip"
+    Write-Output "Downloading Metrics binaries"
+    Start-ExecuteWithRetry { Invoke-WebRequest -Uri $MetricsWindowsBinariesURL -OutFile $filesPath }
+    Write-Output "Extracting binaries archive in: $METRICS_DIR"
+    Expand-Archive -LiteralPath $filesPath -DestinationPath $METRICS_DIR
+ 
+    # Populate the required Metrics service files to the expected locations
+    # This is to match what the current dcos-metrics code base expects
+    # We might need to adjust the locations for those files once we determine 
+    # better locations for them
+    $mesosBin = Join-Path $env:SystemDrive "mesos\bin"
```

Comment:
I would totally love to see these `$mesosBin` and `$mesosVarLib` parameterized. But I think, we have to live with these for now.

Reply:
we will have to finalize the locations for those two files (in dcos-go) before the  Packagepanda is completed.  Once it's determined, we can update the location and parameter them